### DISCUSSION TITLE
Use plugin 'stand-alone'

### DISCUSSION THREAD

**Discussion by edsub** [jobvk/Home-Assistant-Windcentrale#74] [Support] [answered]
After the shut-down of the previous (open) API that Windentrale offered, I found this one that seems to be based on the new "Mijn Windentrale" environment.
As I do not use Home Assistant (but Domoticz) I am looking for ways to run the code of this plugin 'standalone' and just 'spit-out' values (that I can then use to populate Domoticz).

Having limited knowledge of Python I think this is do-able, but I am completely unfamiliar with e.g. the configuration parts that are currently filled in by Home Assistant. I'd prefer to create a config file or the like with e.g. my credentials that the script will then use to authenticate me before collecting 'my' data.

Any suggestion? Or smeone that has worked on the same subject?

**Comment by jobvk** [comment, marked as answer]
The username parameter you should fill in your email address and not a username.

But for testing credentials:
``` python
import json
import requests
import boto3
from pycognito.aws_srp import AWSSRP

boto3_client = boto3.client('cognito-idp', region_name='eu-west-1')
aws = AWSSRP(username='', password='', pool_id='eu-west-1_U7eYBPrBd', client_id='715j3r0trk7o8dqg3md57il7q0', client=boto3_client)
tokens = aws.authenticate_user()
token_type = tokens['AuthenticationResult']['TokenType']
id_token = tokens['AuthenticationResult']['IdToken']
authorization_header = {'Authorization':token_type + " " + id_token}

Request_result = requests.get("https://mijn.windcentrale.nl/api/v0/livedata?projects=WND-WJ", headers=authorization_header, verify=True)
print(Request_result.json())
```

**Comment by edsub** [comment]
Thx, much appreciated. Meanwhile I also found a (similar) script from Marcel Verpaalen.
This will definitely work for me. Planning to fetch data every 5 or 15 minutes and log that (in Domoticz).
Looking forward when Windcenrale will include live production figures again. 
Currentle we seem to have to work with just the totals, or is there an api to obtain historic values as well?
